src/formElementPlugins/dropdownChoice/src/ApplicationView.tsx

```diff
@@ -1,47 +1,67 @@
-import React, { useEffect } from 'react'
+import React, { useEffect, useState } from 'react'
 import { Dropdown, Header, Label } from 'semantic-ui-react'
 import { ApplicationViewProps } from '../../types'
 
 const ApplicationView: React.FC<ApplicationViewProps> = ({
   parameters,
   onUpdate,
-  value,
-  setValue,
+  // value,
+  // setValue,
   isEditable,
   currentResponse,
   validationState,
   onSave,
   Markdown,
   getDefaultIndex,
 }) => {
-  const { label, description, placeholder, search, options, default: defaultOption } = parameters
+  const {
+    label,
+    description,
+    placeholder,
+    search,
+    options,
+    optionsDisplayProperty,
+    default: defaultOption,
+  } = parameters
+
+  const [selectedIndex, setSelectedIndex] = useState<number>()
 
   useEffect(() => {
-    onUpdate(value)
     // This ensures that, if a default is specified, it gets saved on first load
-    if (!value && defaultOption !== undefined) {
-      const defaultIndex = getDefaultIndex(defaultOption, options)
-      const defaultValue = options?.[defaultIndex]
+    if (!currentResponse?.text && defaultOption !== undefined) {
+      const optionIndex = getDefaultIndex(defaultOption, options)
       onSave({
-        text: defaultValue,
-        optionIndex: defaultIndex,
+        text: optionsDisplayProperty
+          ? options[optionIndex][optionsDisplayProperty]
+          : options[optionIndex],
+        selection: options[optionIndex],
+        optionIndex,
       })
-      setValue(defaultValue)
+      setSelectedIndex(optionIndex)
+    }
+    if (currentResponse?.text) {
+      const { optionIndex } = currentResponse
+      setSelectedIndex(optionIndex)
     }
   }, [])
 
   function handleChange(e: any, data: any) {
-    const { value } = data
-    setValue(value)
-    onSave({ text: value, optionIndex: options.indexOf(value) })
+    const { value: optionIndex } = data
+    setSelectedIndex(optionIndex)
+    onSave({
+      text: optionsDisplayProperty
+        ? options[optionIndex][optionsDisplayProperty]
```

Comment:
Same problem here. I think you need something like 
```
const fullObject = options[optionIndex]
return fullObject[optionsDisplayProperty] || fullObject
```

Reply:
Same comment as earlier -- we don't want to be saving a "fullObject" to a text value -- that would cause worse problems elsewhere as "text" is expected to be a string throughout the app. Better to just have the plugin fail and show error, don't you think?

Reply:
Yes, I agree. I was just trying to see, now I got it. Maybe a good idea to flag that on the documentation (if you haven't already).

Reply:
Sure, I'll add a quick note to the docs before I merge this. Thanks for review